### PR TITLE
#163426497 Articles search functionality

### PR DESCRIPTION
**Pull request by ozone4real**
#### What does this PR do?
It implements articles search and filter functionality
#### How should this be manually tested?
Clone this repo and run `npm test`

#### What are the relevant pivotal tracker stories?
[#163426497](https://www.pivotaltracker.com/story/show/163426497)
#### Screenshots (if appropriate)
Screen Shot 2019-04-03 at 2 40 20 PM

Screen Shot 2019-04-03 at 7 10 26 PM

Screen Shot 2019-04-03 at 7 10 08 PM

### CODE REVIEW

**Review by EfeAgare — DISMISSED** [review]

**Review by omenkish — DISMISSED** [review]

**Review by EfeAgare — APPROVED** [review]

**Review by omenkish — APPROVED** [review]

File: __test__/components/comments/__snapshots__/CommentItem.test.js.snap [hunk 1]
```diff
@@ -1,319 +0,0 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (p-funky) [inline __test__/components/comments/__snapshots__/CommentItem.test.js.snap:1]: Why was this file deleted?

File: __test__/components/comments/CommentItem.test.js [hunk 2]
```diff
@@ -17,30 +17,33 @@ const props = {
   isLoading: false
 };
 
-test('It should render the comment item component', () => {
-  const wrapper = shallow(<CommentItem comment={comments[0]} {...props}/>);
-  expect(wrapper).toMatchSnapshot();
-});
```
Comment (p-funky) [inline __test__/components/comments/CommentItem.test.js:23]: Why did you remove this snapshot?

Comment (ozone4real) [reply __test__/components/comments/CommentItem.test.js:23]: It was failing on Travis.

Comment (p-funky) [reply __test__/components/comments/CommentItem.test.js:23]: Not a good way of doing things man... at least not without calling my attention to it.
